app/src/main/java/org/sopt/dosopttemplate/SignUpActivity.kt

```diff
@@ -41,51 +44,60 @@ class SignUpActivity : AppCompatActivity() {
         }
     }
 
-    private fun canSignUp() {
-        if (binding.etSignUpId.length() in 6..10 && binding.etSignUpPw.length() in 8..12
-            && !binding.etSignUpName.text.isNullOrBlank() && !binding.etSignUpMbti.text.isNullOrBlank()
-        ) {
-            intent = Intent(this, LoginActivity::class.java)
-                .putExtra("id", binding.etSignUpId.text.toString())
-                .putExtra("pw", binding.etSignUpPw.text.toString())
-                .putExtra("nickname", binding.etSignUpName.text.toString())
-                .putExtra("mbti", binding.etSignUpMbti.text.toString())
-
-            setResult(RESULT_OK, intent)
-            Toast.makeText(this, "회원가입에 성공했습니다.", Toast.LENGTH_SHORT).show()
-
-            finish()
-        } else
-            Toast.makeText(this, "회원가입에 실패했습니다.", Toast.LENGTH_SHORT).show()
+    private fun checkIdValid() {
+        authSignUpViewModel.username.observe(this) { id ->
+            Log.e("id", "${id}")
+            if (id.isNullOrBlank() || authSignUpViewModel.isIdValid()) {
+                binding.tvSignUpId.error = null
+            } else if (!authSignUpViewModel.isIdValid()) {
+                binding.tvSignUpId.error = "영문, 숫자를 포함해 6-10자 이내로 쓰십시오"
+            }
+        }
```

Comment:
isIdValid() 함수의 반환값을 두 번 확인하고 있네요 ..!
 ```suggestion
private fun checkIdValid() {
    authSignUpViewModel.username.observe(this) { id ->
        if (!id.isNullOrBlank() && !authSignUpViewModel.isIdValid()) {
            binding.tvSignUpId.error = "영문, 숫자를 포함해 6-10자 이내로 쓰십시오"
        } else {
            binding.tvSignUpId.error = null
        }
    }
}
```
아래 checkPwValid도 마찬가지입니다!

Reply:
코드가 훨 좋아보이네요 !! 바로 반영하겠습니다